detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleSetProviderCollector.kt

```diff
@@ -90,17 +94,29 @@ class RuleSetProviderVisitor : DetektVisitor() {
 
         if (expression.calleeExpression?.text == "RuleSet") {
             val ruleListExpression = expression.valueArguments
-                    .map { it.getArgumentExpression() }
-                    .firstOrNull { it?.referenceExpression()?.text == "listOf" }
+                .map { it.getArgumentExpression() }
+                .firstOrNull { it?.referenceExpression()?.text == "listOf" }
                 ?: throw InvalidDocumentationException("RuleSetProvider $name doesn't provide list of rules.")
 
             val ruleArgumentNames = (ruleListExpression as? KtCallExpression)
-                    ?.valueArguments
-                    ?.map { it.getArgumentExpression() }
-                    ?.mapNotNull { it?.referenceExpression()?.text }
+                ?.valueArguments
+                ?.map { it.getArgumentExpression() }
+                ?.mapNotNull(this::guessRuleNameOrNull)
                 ?: emptyList()
 
             ruleNames.addAll(ruleArgumentNames)
         }
     }
+
+    private fun guessRuleNameOrNull(expressionOrNull: KtExpression?): String? {
```

Comment:
```suggestion
    private fun guessRuleNameOrNull(expressionOrNull: KtExpression): String? {
```
There's not reason to accept a `null` value here, right?